index.js

```diff
@@ -35,41 +61,27 @@ DB.prototype.configure = function (config) {
 };
 
 /**
- * Run DB query, it uses prepared statements
+ * Run a DB query using prepared statements. This function does not support batch
+ * operations
  * @param  {String} query
  * @param  {Object} [params]
  * @return {Promise}
  */
-DB.prototype.query = function (query, params) {
-  let connection;
-
-  return this.getConnection().then((conn)=> {
-    console.log('getting connection');
-    connection = conn;
-    return connection.execute(query, params);
-  }).then((results) => {
-    if (connection && connection.connection) {
-      connection.connection.unprepare(query);
-      connection.release();
-      console.log('ok - released connection');
-    }
-
-    return results[0];
-  }).catch(err => {
-    if (!connection) {
-      console.error('could not establish db connection', err.message);
-      throw err;
-    }
-    if (connection.connection) {
-      connection.connection.unprepare(query);
-    }
+DB.prototype.query = function(query, params) {
+  return runQueryWith('execute', query, params);
+}
 
-    connection.release();
-    console.error(`released connection, even with error in sql query execution. Error Message : ${err.message} - `, query, params);
 
-    throw err;
-  });
-};
+/**
+ * Run a DB query without using prepared statements. This function supports batch
+ * operations
+ * @param  {String} query
+ * @param  {Object} [params]
+ * @return {Promise}
+ */
+DB.prototype.bulk = function(query, params) {
+  return runQueryWith('query', query, params);
```

Comment:
As we discussed, `query` is already doing that for us, so there is no need to do it again.